src/components/ClipboardButton/ClipboardButton.tsx

```diff
@@ -49,20 +55,35 @@ const ClipboardButtonComponent = (props: ClipboardButtonComponentProps) => {
         extraProps = {},
         children,
         iconPosition = 'start',
+        timeout,
         ...rest
     } = props;
 
     const buttonIcon = (
-        <Button.Icon>
+        <Button.Icon className={b('icon')}>
             <ClipboardIcon size={ButtonSizeToIconSize[size]} status={status} />
         </Button.Icon>
     );
 
+    const [tooltipTitle, setTooltipTitle] = React.useState(tooltipInitialText);
+    const [tooltipCloseDelay, setTooltipCloseDelay] = React.useState<number | undefined>(undefined);
```

Comment:
Delay may be calculated during render:
`closeDelay={status === 'success' ? timeout : 0}`